openstack/nova/templates/placement-db-unsplit-job.yaml

```diff
@@ -0,0 +1,48 @@
+{{- if and .Values.placement.enabled .Values.placement.unsplit }}
+{{- /* See: https://github.com/openstack/placement/blob/master/placement_db_tools/mysql-migrate-db.sh */}}
+{{- $envAll := . }}
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{ .Release.Name }}-placement-db-unsplit"
+  labels:
+    app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+    app.kubernetes.io/instance: {{ .Release.Name | quote }}
+    app.kubernetes.io/version: {{ default "unknown" .Chart.AppVersion }}
+    helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+  annotations:
+    "helm.sh/hook": pre-upgrade
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  template:
+    metadata:
+      name: "{{ .Release.Name }}-placement-db-unsplit"
+      labels:
+        app.kubernetes.io/managed-by: {{ .Release.Service | quote }}
+        app.kubernetes.io/instance: {{ .Release.Name | quote }}
+        helm.sh/chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    spec:
+      restartPolicy: Never
+      serviceAccountName: {{ .Release.Name }}-pod-exec
+      containers:
+      - name: placement-db-unsplit
+        image: "{{ required ".Values.global.dockerHubMirrorAlternateRegion is missing" .Values.global.dockerHubMirrorAlternateRegion }}/{{ .Values.pre_change_hook.image }}:{{ .Values.pre_change_hook.image_version }}"
+        command:
+        - kubectl-v{{ .Values.pre_change_hook.kubectl_version }}
+        - exec
+        - -c
+        - mariadb
+        - deploy/nova-api-mariadb
+        - --
+        - mariadb
+        - -uroot
+        - --batch
+        - -e
+        - |
+          {{- range mustReverse .Values.placement.tables }}
+            {{- list $envAll . | include "helpers.table_dealias" | indent 10 }}
+          {{- end }}
+
+          DROP DATABASE IF EXISTS {{ .Values.placementdbName }};
+          DROP USER IF EXISTS {{ .Values.placementdbUser }};
```

Comment:
We don't check for those variables to be set, while we do check for them when splitting. Is it save to run this with null values or empty strings?

Reply:
Yes, it results in a syntax error.
> Query Error: Error: ER_PARSE_ERROR: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1